### PR TITLE
Add monthly report for August 2022

### PR DESCRIPTION
**Pull request by philberty**

### CODE REVIEW

**Comment by philberty** [comment]
At some point we should write a small section about the overflow checks here @CohenArthur no rush though

File: 2022-08-monthly-report.org [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+| Data Structures 1 - Core          |       100% |       100% | -     | 30th Nov 2020  | 27th Jan 2021   | 29th Jan 2021 |
+| Control Flow 1 - Core             |       100% |       100% | -     | 28th Jan 2021  | 10th Feb 2021   | 26th Feb 2021 |
+| Data Structures 2 - Generics      |       100% |       100% | -     | 11th Feb 2021  | 14th May 2021   | 28th May 2021 |
+| Data Structures 3 - Traits        |       100% |       100% | -     | 20th May 2021  | 17th Sept 2021  | 27th Aug 2021 |
+| Control Flow 2 - Pattern Matching |       100% |       100% | -     | 20th Sept 2021 | 9th Dec 2021    | 29th Nov 2021 |
+| Macros and cfg expansion          |       100% |       100% | -     | 1st Dec 2021   | 31st Mar 2022   | 28th Mar 2022 |
+| Imports and Visibility            |       100% |       100% | -     | 29th Mar 2022  | 13th Jul 2022   | 27th May 2022 |
+| Const Generics                    |        45% |        75% | +30%  | 30th May 2022  | -               | 17th Oct 2022 |
+| Intrinsics and builtins           |         0% |         0% | -     | 6th Sept 2022  | -               | 14th Nov 2022 |
```
Comment (CohenArthur) [inline 2022-08-monthly-report.org:130]: We should add the intrinsics milestone percentage here as well

**Review by CohenArthur — APPROVED** [review]
Nice one :) I'll add bits about the overflow traps as well as the testing project right away

File: 2022-08-monthly-report.org [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+- Bugs
+ 
+** Detailed changelog
+
+*** Libcore SIP Hasher
+
+We reached a milestone where we can fully compile one of our goal test cases libcore SIP Hasher taken from https://github.com/rust-lang/rust/blob/master/library/core/src/hash/sip.rs
+
+compiler explorer example: https://godbolt.org/z/bn4s54v67
+github-issue: https://github.com/Rust-GCC/gccrs/issues/1247
```
Comment (EdorianDark) [inline 2022-08-monthly-report.org:152]: The bug seems to be still in progress. Or maybe it wasn't updated.

File: 2022-08-monthly-report.org [hunk 3]
```diff
@@ -0,0 +1,454 @@
...
+*** Integration tests update
+
+We have a hit a break through we our blake3 integration test, so that most of the code now compiles. The remaining issues we have our missing for-loop support and some minor bugs in our range syntax to finish this off. For loop's sound simple but they actualy depend on a tremendous amount of libcore code, but the benifit here is that once you support for loops you implicitly support iterators from libcore because a for loop is syntactic sugar for calling IntoIterator and calling next with a check on the result on whether to break or not. If you are interested in what this means, try compiling an empty for loop on compiler explorer for Rustc and see how much code is produced when optimizations are turned off to see what we mean.
+
+see:
+
+- https://godbolt.org/z/hhKnnb3Wo
+- https://godbolt.org/z/PzxEr9vos
+- https://github.com/Rust-GCC/gccrs/issues/682
+- https://github.com/Rust-GCC/gccrs/issues/1247
```
Comment (EdorianDark) [inline 2022-08-monthly-report.org:386]: Already mentioned in line 152

File: 2022-08-monthly-report.org (hunk 2 above)
Comment (philberty) [reply 2022-08-monthly-report.org:152]: There are some subtasks there that need updated still so its not time to close the issue yet, but it does need an update.

File: 2022-08-monthly-report.org (hunk 3 above)
Comment (philberty) [reply 2022-08-monthly-report.org:386]: good catch

**Comment by CohenArthur** [comment]
@EdorianDark how does that sound?

**Comment by EdorianDark** [comment]
Sounds fine!

**Pull request merged**